crates/nu-parser/src/hir.rs

```diff
@@ -467,16 +491,20 @@ impl PrettyDebugWithSource for SpannedLiteral {
             Literal::Size(number, unit) => {
                 b::typed("size", (number.pretty() + unit.pretty()).group())
             }
+            #[allow(deprecated)]
```

Comment:
Before we go this direction, let's chat a bit more first. It looks like there's a lot going on in this PR and it might make sense to do a smaller amount of work first, make sure that lands okay, then grow from there.

We'd be happy to chat on discord to brainstorm about approaches.